faker/providers/person/pl_PL/__init__.py

```diff
@@ -774,3 +788,26 @@ def pwz_nurse(self, kind='nurse'):
         kind_char = 'A' if kind == 'midwife' else 'P'
 
         return '{:02d}{}{}'.format(region, ''.join(map(str, core)), kind_char)
+
+    def nip(self):
+        """
+        Returns 10 digit of Number of tax identification.
+        Polish: Numer identyfikacji podatkowej (NIP).
+
+        https://pl.wikipedia.org/wiki/NIP
+
+        """
+        nip = [self.random_digit() for _ in range(9)]
+
+        weights = (6, 5, 7, 2, 3, 4, 5, 6, 7)
+        check_sum = sum(d * w for d, w in zip(nip, weights)) % 11
+
+        if check_sum % 11 == 10:
+            position = self.random_int(0, 8)
+            nip[position] = (nip[position]+1) % 10
```

Comment:
> Increment/decrement a random digit by 1 (whichever is possible)

If you look at one of my comments, this is why I said "whichever is possible". Your code is failing, because you took a shortcut here. What this line does is essentially loop around from 9 back to 0. If you do that, you effectively performed a decrement of 9, and the check sum actually decreases by 9 times the weight for the digit.